lib/internal/url.js

```diff
@@ -436,8 +436,9 @@ ObjectDefineProperties(URL.prototype, {
             ret += `:${ctx.password}`;
           ret += '@';
         }
+        const hostname = this[context].host || '';
```

Comment:
```suggestion
```

This condition shouldn't be necessary as we already are in `if (ctx.host !== null)`